src/device.cc

```diff
@@ -281,33 +282,117 @@ std::shared_ptr<SubHidrawConnection> SubHidrawConnection::create(const DeviceSca
 
   connection->m_details.phys = sd.phys;
   connection->disableWrite(); // disable write notifier
+  connection->initSubDevice();
   return connection;
 }
 
 // -------------------------------------------------------------------------------------------------
-ssize_t SubDeviceConnection::sendData(const QByteArray& hidppMsg)
+void SubDeviceConnection::initSubDevice()
+{
+  struct timespec ts;
+  int msec = 50;
+  ts.tv_sec = msec / 1000;
+  ts.tv_nsec = (msec % 1000) * 1000000;
+
+  resetSubDevice(ts);
+
+  // Add other configuration to enable features in device
+  // like enabling on Next and back button on hold functionality.
+  // No intialization needed for Event Sub device
+}
+
+// -------------------------------------------------------------------------------------------------
+void SubDeviceConnection::resetSubDevice(struct timespec delay)
+{
+  // Ping spotlight device for checking if is online
+  pingSubDevice();
+  ::nanosleep(&delay, &delay);
```

Comment:
I am not happy with sleeps in code - putting the main thread to sleep is never a good idea.

Also: why 50 ms?

Reply:
I have some unallocated time on Tuesday/Wednesday - if you can wait until then - I could merge into a feature branch and refactor the sleep code.

Reply:
Used `QTimer::singleShot` in the latest commit.